dagshub/data_engine/model/datasource.py

```diff
@@ -334,6 +334,38 @@ def as_of(self, time: Union[float, datetime.datetime]) -> "Datasource":
         new_ds._query.as_of = to_timestamp(time)
         return new_ds
 
+    def order_by(self, *args: Union[str, Tuple[str, bool], Tuple[str, str]]):
+        """
+        Sort the query result by the specified fields.
+
+        Args:
+            Fields to sort by. Can be either of:
+                - Name of the field to sort by: ``"field"``.
+                - A tuple of ``(field_name, ascending)``: ``("field", True)``.
+                - A tuple of ``(field_name, "asc"|"desc")``: ``("field", "asc")``.
+
+        Examples::
+
+            ds.sort_by("size", ("date", "desc")).all()
+            ds.sort_by("size", ("date", False)).all()
+        """
+        new_ds = self.__deepcopy__()
+        new_ds.get_query().order_by = []
```

Comment:
Conflicted on whether clearing is a good idea. I imagine a user that doesn't know you could order_by multiple columns might do
```python
ds.order_by("size").order_by("is_val")
```

On the other hand if you don't reset it here, then there's no way to reset a sort 🤔 

You should 100% say that calling the function resets whatever sort was set previously
By the way, you should also say that doing an `ds.order_by()` resets the sort to the default (datapoint addition order)